### PR TITLE
EXPANDR-1576 CortexXpanse Remediation Guidance changes

### PR DESCRIPTION
**Pull request by capanw**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/EXPANDR-1576

## Description
Adding Attack surface rule command to the integration

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by capanw** [comment]
@johnnywilkes  @BigEasyJ When you guys get a chance, can you review.

**Review by johnnywilkes — APPROVED** [review]
@capanw , LGTM

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py [hunk 1]
```diff
@@ -140,6 +140,25 @@ def get_external_ip_address_range_request(self, range_id_list: list[str]) -> dic
 
         return response
 
+    def get_attack_surface_rule_request(
+        self, search_params: List[dict]
```
[the remark this answers is not in the document]
Comment (capanw) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py:144]: @BigEasyJ Wanted to keep the code consistent/similar with other function definitions and make minimal changes to the code.

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse_test.py [hunk 2]
```diff
@@ -62,6 +62,47 @@ def test_list_external_service_command(requests_mock):
...
+        base_url='https://test.com/api/webapp/public_api/v1/',
+        verify=True,
+        headers={
+            "HOST": "test.com",
+            "Authorization": "THISISAFAKEKEY",
+            "Content-Type": "application/json"
+        },
+        proxy=False)
+
+    args = {
```
[the remark this answers is not in the document]
Comment (capanw) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse_test.py:95]: I don't think so. I am mocking the response from get_attack_surface_rule_command so didn't bother making any changes.

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py [hunk 3]
```diff
@@ -556,6 +575,65 @@ def get_external_ip_address_range_command(client: Client, args: dict[str, Any])
...
+            "field": "category",
+            "operator": "in",
+            "value": category.split(",")
+        })
+
+    response = client.get_attack_surface_rule_request(search_params)
+    parsed = response.get("reply", {}).get("attack_surface_rules")
+    command_results = CommandResults(
+        outputs_prefix='ASM.AttackSurfaceRule',
+        outputs_key_field='attack_surface_rule',
```
Comment (ilappe) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py:629]: it's look like `attack_surface_rule` didn't defined in the YML file and. in the results
do you mean `attack_surface_rule_id`

**Review by johnnywilkes — APPROVED** [review]
LGTM

**Comment by ilappe** [comment]
Hi @capanw 
changes like changing the command name are breaking changes (BC)
and need a good reason why to do that 

so the validation fail on [BC validation](https://app.circleci.com/pipelines/github/demisto/content/319357/workflows/b9fdaeda-4d2e-495e-9954-562d7abd14b0/jobs/730013?invite=true#step-119-7217_92)

if you're sure that this change should be merged, you should create the release notes with the `-bc` flag
see [breaking changes version](https://xsoar.pan.dev/docs/documentation/release-notes#breaking-changes-version)

**Comment by capanw** [comment]
Hello @ilappe, Thanks for reviewing. We are updating a command name _asm-list-attack-surface-rules_ to _asm-get-attack-surface-rule_. We wanted to keep this command the same as the command in CortexAttackSurfaceMangement pack. For Backward compatibility, I checked all through the packs and did not find any packs thats using _asm-list-attack-surface-rules_. So, there should not be anything thats would break from what I looked into.

I updated the release notes according to your suggestions.

**Comment by ilappe** [comment]
Hi @capanw 
think about customer, which already use this pack and build a playbook using the command `asm-list-attack-surface-rules`
the change here will break it's playbook 
so we should avoid from changes like this and this are the reason why the [pre-commit ](https://github.com/demisto/content/actions/runs/6946572567/job/18898383037?pr=30712) failed.

In case you decide that this change is necessary, you need to create a BC release notes (by runing the command with `-bc` as explained here [breaking changes version](https://xsoar.pan.dev/docs/documentation/release-notes#breaking-changes-version)) and as a result, a json file like [this](https://github.com/demisto/content/blob/master/Packs/CortexXDR/ReleaseNotes/5_0_0.json) should be created, with `breakingChangesNotes` key, where you specify what is broken.

**Comment by capanw** [comment]
Hello @ilappe, thank you for providing those links. Makes sense and agreed on the customer part. We wanted to make sure both the packs CortexXpanse and CortexAttackSurfaceManagement are in sync. Hence making these changes. We will try and reduce these breaking changes in the future.

**Review by ilappe — APPROVED** [review]
Ok,
Thank you!

**Pull request merged**